version: '3'
services:
  django:
    build:
      context: .
      dockerfile: Dockerfile
    image: sec-analyzer-server
    container_name: django
    restart: unless-stopped
    env_file: .env
    environment:
      - DB_HOST=postgresqldb
      - CHOKIDAR_USEPOLLING=true
    ports:
      - 7860:7860
    volumes:
      - .:/app
    depends_on:
      - postgresqldb
    networks:
      - app-network
    command:
      - /bin/sh
      - -c
      - |
        python manage.py migrate
        python manage.py runserver 0.0.0.0:7860
  postgresqldb:
    image: 'bitnami/postgresql:latest'
    container_name: postgresqldb
    restart: unless-stopped
    env_file: .env
    environment:
      - POSTGRESQL_USERNAME=$DB_USER
      - POSTGRESQL_PASSWORD=$DB_PASSWORD
      - POSTGRESQL_DATABASE=$DB_NAME
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U $DB_USER"]
      interval: 10s
      timeout: 5s
      retries: 5
    volumes:
      - dbdata:/bitnami/postgresql
    networks:
      - app-network

networks:
  app-network:
    driver: bridge

volumes:
  dbdata:
